de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ft/FaultTreeDiagramToolBehaviorProvider.java

```diff
@@ -118,4 +128,66 @@ public Object getToolTip(GraphicsAlgorithm ga) {
 		
 		return super.getToolTip(ga);
 	}
+	
+	@Override
+	public IPaletteCompartmentEntry[] getPalette() {
+		final String SIGNATURE = "getPaletteCompartments()";
+		boolean info = T.racer().info();
+		if (info) {
+			T.racer().entering(DefaultToolBehaviorProvider.class, SIGNATURE, new Object[0]);
+		}
+		List<IPaletteCompartmentEntry> compartments = new ArrayList<IPaletteCompartmentEntry>();
+
+		PaletteCompartmentEntry compartmentEntry = new PaletteCompartmentEntry("Connections", null);
+		compartments.add(compartmentEntry);
+
+		IFeatureProvider featureProvider = getFeatureProvider();
+		ICreateConnectionFeature[] createConnectionFeatures = featureProvider.getCreateConnectionFeatures();
+		if (createConnectionFeatures.length > 0) {

```

Comment:
I dont think we need this if check. If the array is empty, then the for loop will exit immediately.

Reply:
That's how it was done in the overwritten method. I was thinking the same.